first attempt at refactoring README (#10742)
spectrocloud/librarium: docs/contributing/markdown-and-mdx.md

when reviewing the examples below:

```shell
.
└── docs
    └── docs-content
        ├── architecture
        │   ├── grpc.md
        │   └── ip-addresses.md
        ├── aws
        │   └── iam-permissions.md
        ├── clusters
        └── security.md
```

### Same Folder

To link to a file in the same folder, you can use the following syntax:

```md
![Insert a description here](name_of_file.md)
```

Because the file is in the same folder, you do not need to specify the path to the file. Docusaurus will automatically
search the current folder for the file when compiling the markdown content.

So, if you are in the file `grpc.md` and want to reference the file `ip-addresses.md`, you would use the following
syntax:

```md
![A list of all Palette public IP addresses](ip-addresses.md)
```

### Different Folder

If you want to link to a file in a different folder, you have to specify the path to the file from where the current
markdown file is located.

If you are in the file `security.md` and want to reference the file `iam-permissions.md`, you have to use the following
syntax:

```md
![A list of all required IAM permissions for Palette](aws/iam-permissions.md)

> ```

Reviewer: This is a bit confusing. Maybe we can slightly reframe this section to say they can use either a relative path or absolute path from the `docs-content` root? And give examples of each (like you have)? Saying, "you have to use the following syntax" isn't per se correct.

### A Heading in a Different File

To link to a heading in a different file, you can use the following syntax:

```md
[Link to a heading in a different file](name_of_file.md#heading-name)
```

For example, if you are in the file `grpc.md` and want to reference the heading `Palette gRPC API` in the file
`security.md`, you would use the following syntax:

```md
[Link to a heading in a different file](../security.md#palette-grpc-api)
```

The important thing to remember is that the `#` comes after the file name and before the heading name.

### Exceptions

As of Docusarus `2.4.1`, the ability to link to documentation pages that belong to another plugin is unavailable. To
work around this limitation, reference a documentation page by the URL path versus the file path.

```md
[Link to a page in another plugin](/api-content/authentication#api-key)
```

> [!WARNING] Be aware that this approach will break versioning. The user experience will be impacted as the user will be
> redirected to the latest version of the page.

In future releases, Docusaurus will support linking pages from other Docusarus plugins. Once this feature is available,

> this documentation will be updated.

Reviewer: Proposing an updated version (thanks robots!) to help clarify why API links are different. The "broken versioning" note was also no longer accurate because of Adelina's changes from way-back-when.

(Putting suggestion below because of nested code blocks. Edit my comment to grab the raw MD.)

Reviewer: ### Linking to and from API Docs

In Docusaurus, a **plugin** is a separate content system that owns its own source files, routes, and sidebar. Each
plugin processes Markdown independently and maintains its own file-path-to-URL mapping.

This repository configures two docs plugins in `docusaurus.config.js`:

| Plugin                 | Source folder                | URL base | Content               |
| ---------------------- | ---------------------------- | -------- | --------------------- |
| Main docs (default)    | `docs/docs-content/`         | `/`      | Product documentation |
| API docs (`id: "api"`) | `docs/api-content/api-docs/` | `/api`   | API reference         |

> [!NOTE] Tutorials and Downloads have their own sidebars in the navbar, but they are **not** separate plugins. They are
> folders within `docs/docs-content/` and use the same URL base (`/`) as the rest of the product docs. The site shows a
> different sidebar depending on which navbar item you select (configured in `sidebars.js` and `docusaurus.config.js`),
> but Docusaurus still processes all of those pages as one plugin. That is why a tutorial can link to a product doc with
> a file path such as `../../profiles/profiles.md`, while a link to the API reference must use a URL path such as
> `/api/introduction`.

File-path links with a `.md` extension only work **within the same plugin**. You can link from one product doc to
another using the file-path syntax described above because both files live under `docs/docs-content/` and are processed
by the main docs plugin.

Docusaurus cannot resolve a `.md` file path across plugins. For example, a link from `docs/docs-content/` to
`docs/api-content/api-docs/1-introduction.md` does not work, even though both folders are under `docs/`. To link between
plugins, use a **URL path** instead. Do not include a file extension in the link.

```md
<!-- From a page in docs/docs-content/ -->

[Palette API introduction](/api/introduction)

[API Key section](/api/introduction#api-key)

[Palette API v1 endpoints](/api/category/palette-api-v1/)
```

The same rule applies in reverse. From a page under `docs/api-content/`, link back to product docs with a URL path:

```md
<!-- From a page in docs/api-content/ -->

[Create API Key](/user-management/authentication/api-key/create-api-key)
```

Use the route path the page appears at on the site, not the folder path in the repository. For example, use
`/api/introduction`, not `/api-content/api-docs/1-introduction.md` or `/api-content/authentication`.

> [!NOTE] Root-relative URL paths such as `/api/introduction` resolve against the current host. On a legacy deployment
> such as [v4.8.x](https://version-4-8.legacy.docs.spectrocloud.com/), that link stays on the versioned site. Avoid
> hard-coded links to `https://docs.spectrocloud.com/...` in content that must work on archived versions, because those
> always point to the latest production site.

## Redirects

> To add a redirect to an existing documentation page you must add an entry to the [redirects.js](../../redirects.js)

Reviewer: ```suggestion
If a file already exists and you need to move it elsewhere, add a redirect entry that points from the old path to the new path using the [redirects.js](../../redirects.js)
```

> file. Below is an example of what a redirect entry should look like.

Reviewer: ```suggestion
file. This ensures that anywhere old links are used, whether that be links embedded in the product or pages saved by customers, continue to work. Below is an example of what a redirect entry should look like.
```

```js
  {
    from: `/clusters/nested-clusters/`,
    to: `/clusters/sandbox-clusters`,
  },
```

## Images or other assets

All images must reside in the [`static/assets/docs/images`](../../static/assets/docs/images/) folder. All images must be

> in webp format. You can save png, jpg, or jpeg to the directory. The commit hook will convert the images to webp format.

Reviewer: ```suggestion
in webp format. If you save png, jpg, or jpeg to the directory, the commit hook will convert the images to webp format.
```

> Or issue the command `make format-images` to convert the images to webp format.

Reviewer: ```suggestion
Alternately, issue the command `make format-images` to convert the images to webp format.
```

```md
![alt text](/clusterprofiles.png "cluster profiles example")
```

You can add a directory to to the images folder.

```md
![alt text](/introduction/clusterprofiles.png "cluster profiles example")
```

**Image Loading** Image size loading can be customised. You can provide eager-load to images in the first fold of the
image with high priority as LCP (Largest contentful Paint) for the page will not be affected

```md
![alt text eager-load](/clusterprofiles.png)
```

## Code Lines Highlighter

> You can highlight specific lines in a block of code by adding **coloredLines** prop.

Reviewer: ```suggestion
You can highlight specific lines in a block of code by using a [metadata string](https://docusaurus.io/docs/markdown-features/code-blocks#highlighting-with-metadata-string) in the header of the code block.
```

> _Example_: ` ```js {2-4,5-7}`. This will color the lines from 2 to 4 and from 5 to 7.

Reviewer: ```suggestion
For example, using ` ```js {2-4,5-7}` will highlight the lines from 2 to 4 and from 5 to 7 in the code block.
```

_Components_:

- `2-4` - lines interval to be colored

> - `,` - separator for different colored lines intervals

Reviewer: ```suggestion
```

Example

> https://docusaurus.io/docs/markdown-features/code-blocks#highlighting-with-comments

Reviewer: ```suggestion
```

### Hide ClipBoard Button

The copy button is shown by default in all code blocks. You can disable the copy button by passing in the parameter

> value `hideClipboard` in the markdown declaration of the code blocks.

Reviewer: ```suggestion
value `hideClipboard` in the markdown declaration of the code blocks. This is useful for example output where the user does not need to copy or use the content.
```

Example ![Example](../../static/assets/docs/images/hide_copy_button_example.webp)

Result

![Result](/static/assets/docs/images/hide_copy_button.webp)

> ## Admonitions - Warning / Info / Tip / Danger / Tech Preview / Further Guidance

Reviewer: Ah cripes, you'll need to resolve the merge conflict I created with this due to :::deprecated. My bad :(

```

**Image Loading** Image size loading can be customised. You can provide eager-load to images in the first fold of the
image with high priority as LCP (Largest contentful Paint) for the page will not be affected

```md
![alt text eager-load](/clusterprofiles.png)
```

## Code Lines Highlighter

You can highlight specific lines in a block of code by adding **coloredLines** prop.

_Example_: ` ```js {2-4,5-7}`. This will color the lines from 2 to 4 and from 5 to 7.

_Components_:

- `2-4` - lines interval to be colored
- `,` - separator for different colored lines intervals

Example

https://docusaurus.io/docs/markdown-features/code-blocks#highlighting-with-comments

### Hide ClipBoard Button

The copy button is shown by default in all code blocks. You can disable the copy button by passing in the parameter
value `hideClipboard` in the markdown declaration of the code blocks.

Example ![Example](../../static/assets/docs/images/hide_copy_button_example.webp)

Result

![Result](/static/assets/docs/images/hide_copy_button.webp)

## Admonitions - Warning / Info / Tip / Danger / Tech Preview / Further Guidance / Deprecated

For guidance on using admonitions in our docs, refer to

> [Spectro Cloud Internal Style Guide: Admonitions/Callouts](https://spectrocloud.atlassian.net/wiki/spaces/DE/pages/1765933057/Spectro+Cloud+Internal+Style+Guide#Admonitions%2FCallouts).

Reviewer 2: ```suggestion
For guidance on using admonitions in our docs, refer to the
[Spectro Cloud Internal Style Guide: Admonitions](../../style-guide.md#admonitions).
```
